backend/fms_core/template_importer/row_handlers/library_qc/library_qc.py

```diff
@@ -40,11 +40,12 @@ def process_row_inner(self, sample_container, measures, process, process_measure
         volume_used = process_measurement['volume_used']
         sample_volume = source_sample_obj.volume
         final_volume = None
-        
+
         if initial_volume is None:
```

Comment:
We need to validate that if initial_volume is none, measured_volume is none too. Otherwise we introduce false information the volume at the time of the submission might not be the same as at the time of the measurement.

Reply:
It's already done? Unless `source_sample_obj.volume` can be None for some reason...

Reply:
Nope if initial_volume is None before you set it, the original measured_volume can still be anything ... You are falling into your own trap already. We need to distinguish the submitted values and the intention behind them from the values in the variables. I want to make sure that if current volume is not set on the template and a measured volume is set, we don't use an internal value that might be different from the internal value at the time of the operation in the lab (we send back an error requesting a current value). To do that i want to prevent users to submit an empty current volume with a not empty measured volume (so we get a valid delta).

Reply:
Ah oops. I missed the part where you said "if initial_volume is none, measured_volume is none too". Basically make sure either both is None, or only measured is None.